pose-detection/src/calculators/low_pass_filter.ts

```diff
@@ -31,10 +31,22 @@ export class LowPassFilter {
 
   constructor(private alpha: number) {}
 
-  apply(value: number): number {
+  apply(value: number, threshold?: number): number {
     let result;
     if (this.initialized) {
-      result = this.alpha * value + (1 - this.alpha) * this.storedValue;
+      if (threshold == null) {
+        // Regular lowpass filter.
+        // result = this.alpha * value + (1 - this.alpha) * this.storedValue;
+        // We need to reformat the formula to be able to conveniently apply
+        // another optional non-linear function to the
+        // (value - this.storedValue) part.
+        result = this.storedValue + this.alpha * (value - this.storedValue);
+      } else {
+        // Add additional non-linearity to cap extreme value.
```

Comment:
Here I would add that the update function is ~x->x close to zero and behaves like sign(x) log(abs(x)) for large magnitudes of x? It might make things clearer for external people.